### ISSUE TITLE
[TotK] Potential memory leak causing yuzu to close itself

### ISSUE THREAD

**Issue by fpi1337** [yuzu-emu/yuzu#10515]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Affected Build(s)

Early Access 3621

### Description of Issue

After posting my problem in discord, I've no decided to open an issue to help the developers to keep an overview over the issues since they tend to get lost in Discord. I've experienced Yuzu closing itself without any error message or such. However it happened after I landed right after a glide, potentially when the game tries to save. I'm using no mods besides fps++30.

With the kind help of @german77, it was suspected that there could be a memory leak.

### Expected Behavior

The emulator shouldn't close itself.

### Reproduction Steps

Just play the game and "hope" it occurs at some point.

### Log File

Will be provided in a minute. 

### System Configuration

CPU:
GPU/Driver:
RAM:
OS:

**Comment by LtqxWYEG** [comment]
I have the same issue. I increased my virtual memory to 64GB and closed all other programs. That seems to "help", but it still occurs.

Mainly when I do something with objects. Like exploding something near them. One time a electric Wizrobe caused the crash. But it could be because it hit an object aswell.

There's a huge spike in virtual memory usage right before the game crashes. GBs in milliseconds.

I do use a bunch of mods, but it happened before I used any too.

I try to get a Nsight Aftermath report. But I neither know what it is nor if it will show anything useful. I'll report back

**Comment by aviationhacker** [comment]
Hi there, just thought I'd comment on this as I found it via Google - I've had the same actually, and frequently. It seems to happen for me when my Committed Memory reaches maximum (so 36.6Gb/36.6GB) and then Yuzu exits. The OS and other applications are only at 10Gb, so it seems the other 26GB is Yuzu doing something, and once it crashes, it again returns to that same 10Gb mark (I've watched it count down and watched it as it crashes), so quite a lot of RAM is being used.

**Comment by LtqxWYEG** [comment]
Yes that is called a memory leak. Yuzu just uses all available Commit space because one function in its code is asking for more and more memory. Mabe it can't access the already allocated memory and thus requests more, who knows.  
In any case; This should be a high priority bug
